ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -3199,19 +3199,24 @@ public Map<Map<String, String>, Partition> loadDynamicPartitions(final LoadTable
           partitionNames.add(Warehouse.makeDynamicPartNameNoTrailingSeperator(details.fullSpec));
         }
       }
-      List<Partition> partitions = Hive.get().getPartitionsByNames(tbl, partitionNames);
-      for(Partition partition : partitions) {
-        LOG.debug("HMS partition spec: {}", partition.getSpec());
-        partitionDetailsMap.entrySet().parallelStream()
-            .filter(entry -> entry.getValue().fullSpec.equals(partition.getSpec()))
-            .findAny().ifPresent(entry -> {
-          entry.getValue().partition = partition;
-          entry.getValue().hasOldPartition = true;
-        });
-      }
-      // no need to fetch partition again in tasks since we have already fetched partitions
-      // info in getPartitionsByNames()
-      fetchPartitionInfo = false;
+      try {
+        List<Partition> partitions = Hive.get().getPartitionsByNames(tbl, partitionNames);
+        for(Partition partition : partitions) {
+          LOG.debug("HMS partition spec: {}", partition.getSpec());
+          partitionDetailsMap.entrySet().parallelStream()
+              .filter(entry -> entry.getValue().fullSpec.equals(partition.getSpec()))
+              .findAny().ifPresent(entry -> {
+            entry.getValue().partition = partition;
+            entry.getValue().hasOldPartition = true;
+          });
+        }
+        // no need to fetch partition again in tasks since we have already fetched partitions
+        // info in getPartitionsByNames()
+        fetchPartitionInfo = false;
+      } catch (HiveException e) {
+        // Failed to fetch partitions in some cases (e.g., partition number limit)
+        LOG.warn("Fetching partitions by names failed.", e);
```

Comment:
i don't see how this change is relevant to the PR, also we should propagate the exception back to the client

Reply:
There are two ways to fetch partitions from hms in `loadDynamicPartitions()`:
1. get partitions by names
2. get partitions one by one
I think it's reasonable to fallback to the second if the first one failed. And we have a warn log to remind client what error happened in getPartitionsByNames call.

Reply:
oh, is the fallback code below? i don't see we use the same `partitionNames` list there.
what would happen if limit is exceeded? we'll catch the exception and try to load 1-by-1?

Reply:
I didn't see any fallback code block. BTW, i think no one will care about the warn log, so client side(spark & hiveserver2) maybe miss this warn exception, and then they maybe lose some partition data.

Therefore, if this PR must change the client behavior, such as the Introduced `partition limit exception` , (see the previous ci http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-5063/1/tests/ ), we need throw this exception. But as i said above, it would better to not to introduce this incompatible behavior if possible.

Reply:
The only way the client realizes that there is a problem (limit exceeded) is if he checks the logs. I think we should throw an exception

Reply:
Reverted this change.

Reply:
Please see comment https://github.com/apache/hive/pull/5063/files#r1528710963